core/app/models/spree/payment.rb

```diff
@@ -43,6 +53,7 @@ class Payment < Spree::Base
     validates :amount, numericality: true
     validates :source, presence: true, if: :source_required?
     validates :payment_method, presence: true
+    validates :state, inclusion: { in: [PENDING, PROCESSING, CHECKOUT, COMPLETED, INVALID, VOID, FAILED] }
```

Comment:
I feel like a method should be used here. So that way people can easily extend payment states if they need to.

Reply:
Very good point Ben. @jgayfer would you mind to address this?